packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -289,7 +289,38 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
     if (!_enabled) {
       return false;
     }
+
     final loggingConstraint = _getLoggingConstraint();
+
+    String? userId;
+
+    Amplify.Hub.listen(HubChannel.Auth, (AuthHubEvent event) async {
+      if (event.type == AuthHubEventType.signedOut ||
+          event.type == AuthHubEventType.userDeleted ||
+          event.type == AuthHubEventType.sessionExpired) {
+        userId = null;
+      }
+      if (event.type == AuthHubEventType.signedIn) {
+        userId = event.payload?.userId;
+      }
+    });
```

Comment:
rebase with feature branch and then 
instead of having `userId` as a method variable use a private filed `_userId` and use the same hub listener we have  at ctor to update the _userId on Auth event